app/code/core/Mage/Catalog/Model/Resource/Category/Flat.php

```diff
@@ -1311,15 +1311,17 @@ public function isInRootCategoryList($category)
      */
     public function getChildren($category, $recursive = true, $isActive = true)
     {
+        $maintable = $this->getMainStoreTable($category->getStoreId());
         $select = $this->_getReadAdapter()->select()
-            ->from($this->getMainStoreTable($category->getStoreId()), 'entity_id')
+            ->from($maintable, 'entity_id')
             ->where('path LIKE ?', "{$category->getPath()}/%");
         if (!$recursive) {
             $select->where('level <= ?', $category->getLevel() + 1);
         }
         if ($isActive) {
             $select->where('is_active = ?', '1');
         }
+        $select->order($maintable.".position ASC");
```

Comment:
Can we include this as part of the fluent call right after the `where` and drop the `$select`? Moving this up above the if statements would be needed.

Reply:
Thanks for your support. I have refactored it.